utils_cv/detection/model.py

```diff
@@ -23,54 +35,59 @@
 from ..common.gpu import torch_device
 
 
-def _get_det_bboxes(
-    pred: List[dict], labels: List[str], im_path: str = None
-) -> List[DetectionBbox]:
-    """ Gets the bounding boxes and labels from the prediction object
+def _get_det_bboxes_and_mask(
+    pred: Dict, labels: List[str], im_path: Union[str, Path] = None
+) -> Dict:
+    """ Gets the bounding boxes and masks from the prediction object.
 
     Args:
         pred: the output of passing in an image to torchvision's FasterRCNN
-        model
-        labels: list of labels
+            or MaskRCNN model
+        labels: list of labels without "__background__".
         im_path: the image path of the preds
-
     Return:
-        a list of DetectionBboxes
+        a dict of DetectionBboxes and masks
     """
-    pred_labels = pred[0]["labels"].detach().cpu().numpy().tolist()
-    pred_boxes = (
-        pred[0]["boxes"].detach().cpu().numpy().astype(np.int32).tolist()
-    )
-    pred_scores = pred[0]["scores"].detach().cpu().numpy().tolist()
-
-    det_bboxes = []
-    for label, box, score in zip(pred_labels, pred_boxes, pred_scores):
-        label_name = labels[label - 1]
-        det_bbox = DetectionBbox.from_array(
-            box,
-            score=score,
-            label_idx=label,
-            label_name=label_name,
-            im_path=im_path,
-        )
-        det_bboxes.append(det_bbox)
 
-    return det_bboxes
+    label_names = np.array(labels)[pred['labels'] - 1]
+    res = {
+        "det_bboxes":
+            DetectionBbox.from_arrays(
+                pred['boxes'].tolist(),
+                score=pred['scores'].tolist(),
+                label_idx=pred['labels'].tolist(),
+                label_name=label_names.tolist(),
+                im_path=im_path,
+            )
+    }
+
+    if "masks" in pred:
+        res["masks"] = pred["masks"].squeeze(1)
+    return res
 
 
 def _apply_threshold(
-    det_bboxes: List[DetectionBbox], threshold: Optional[float] = 0.5
-) -> List[DetectionBbox]:
-    """ Filters the list of DetectionBboxes by score threshold. """
-    return (
-        [det_bbox for det_bbox in det_bboxes if det_bbox.score > threshold]
-        if threshold is not None
-        else det_bboxes
-    )
+    pred: Dict,
```

Comment:
Similar to the comment above, apply_threshold() in it's current form does more than just applying threshold: it also calls detach() etc. Now this function always has to be called. Please revert back to before where it just returns a subset.

Reply:
I believe we probably want to get rid of "mask_threshold". "threshold" makes sense since it removes a detection is the confidence in the FULL detection is lower than a threshold (not just in the mask)

Reply:
Do you mean to hard-code `mask_threshold` to `0.5` where it appears?

Reply:
I hard-coded `mask_threshold` to `0.5`

Reply:
Thanks. This seems like a good approach for now to clean up this PR.